### PR TITLE
Use PRNG to generate unique pad name

### PR DESCRIPTION
**Pull request by drebs**
This modification uses the same reasoning as the modification made in
Etherpad Lite random pad name generation:

  ether/etherpad-lite#3516
  ether/etherpad-lite#3517

Closes: #2

### CODE REVIEW

**Comment by drebs** [comment]
**Important:** I have not tested this modification, as I don't have the proper development env to do it. Please, test and give feedback before merging. :-)

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ep_new_pad",
-  "version": "0.0.1",
-  "description": "Etherpad lite plugin to crate a REALLY new pad (database check). Call http://example.com/randomPad to use it.",
+  "version": "0.0.2,
```
Comment (Gared) [inline package.json]: Here is one " missing. The rest looks good to me :-)

File: index.js [hunk 2]
```diff
@@ -6,38 +6,32 @@ exports.registerRoute = function (hook_name, args, cb) {
...
+      // use Node.js PRNG to generate a "unique" sequence
+      const crypto = require('crypto');
+      crypto.randomFillSync(randomArray);
       var randomstring = '';
       for (var i = 0; i < length; i++)
       {
-        var charNum = Math.floor(Math.random() * chars.length);
+        // instead of writing "Math.floor(randomarray[i]/256*64)"
+        // we can save some cycles.
+        var charNum = Math.floor(randomarray[i]/4);
```
Comment (Gared) [inline index.js]: random**A**rray

File: index.js [hunk 3]
```diff
@@ -6,38 +6,32 @@ exports.registerRoute = function (hook_name, args, cb) {
...
+      // the length of the pad name is chosen to get 120-bit security:
+      // log2(64^20) = 120
+      var string_length = 20;
+      // make room for 8-bit integer values that span from 0 to 255.
+      var randomarray = new Uint8Array(string_length);
+      // use Node.js PRNG to generate a "unique" sequence
+      const crypto = require('crypto');
+      crypto.randomFillSync(randomArray);
       var randomstring = '';
       for (var i = 0; i < length; i++)
```
Comment (Gared) [inline index.js]: **string_**length?

File: index.js [hunk 4]
```diff
@@ -6,38 +6,32 @@ exports.registerRoute = function (hook_name, args, cb) {
...
-      var chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
-      var length = Math.floor((Math.random()*15)+8);
+      // the number of distinct chars (64) is chosen to ensure that
+      // the selection will be uniform when using the PRNG below
+      var chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-_";
+      // the length of the pad name is chosen to get 120-bit security:
+      // log2(64^20) = 120
+      var string_length = 20;
+      // make room for 8-bit integer values that span from 0 to 255.
+      var randomarray = new Uint8Array(string_length);
```
Comment (Gared) [inline index.js]: random**A**rray

File: package.json (hunk 1 above)
Comment (drebs) [reply package.json]: oops! fixed in 9aa09f2f002f9f3885b317220910fa4041a958be

File: index.js (hunk 2 above)
Comment (drebs) [reply index.js]: I left it with lowercase to match the code in: https://github.com/ether/etherpad-lite/commit/39fbc37dd8f93c1d57d1b2b1fbe8e5feb83d9849

It's fixed in 9aa09f2f002f9f3885b317220910fa4041a958be.

File: index.js (hunk 3 above)
Comment (drebs) [reply index.js]: Fixed in 9aa09f2f002f9f3885b317220910fa4041a958be.

File: index.js (hunk 4 above)
Comment (drebs) [reply index.js]: I left it with lowercase to match the code in: https://github.com/ether/etherpad-lite/commit/39fbc37dd8f93c1d57d1b2b1fbe8e5feb83d9849

It's fixed in 9aa09f2f002f9f3885b317220910fa4041a958be.

**Comment by drebs** [comment]
Thanks for looking into this, and sorry again to not be able to test it. If there's anything else to fix or improve just let me know!

**Comment by Gared** [comment]
Thanks for your work, drebs :+1:

**Pull request merged**